index.rst

```diff
@@ -2,24 +2,180 @@
+
+.. , since we execute (or plan to) these tests continuously, we can verify that the system continues to fullfil requirements at every iteration.
+
+.. Because these functionalities are (mostly) taken from the requirements pool, it means that each new feature is generally accompanied by a requirement verification test.
+.. Nevertheless, tests comes in a variety of different shapes and forms and, most importantly, sometimes it is not possible to fully exercise a given requirement in a testing environment.
+
+.. _verification-artifacts:
+
+Verification artifacts
+======================
+
+In general, successfully executing the testing suite from start to finish gives us confidence that the system is operating as expected.
+Nevertheless, the SAL Scripts used during these tests are only designed to fail execution if some critical condition happen (e.g. a component unexpectedly went to FAULT state), and are not designed to verify all the details of every single component involved in the process.
+
+For example, the ``set_summary_state`` SAL Script is designed to execute state transitions and is commonly used in operations to enable components, recover from fault states, etc.
+They are executed as an initial step during testing to enable all components and, while they verify the components do transition successfully to the desired state, they **do not** check that the CSCs publish all the required events.
+This type of verification need to be done a posteriori and included in verification artifacts.
+
+These artifacts are human readable reports that contain enough information for a reader with some knowledge about the system can understand what is being verified and how.
+For instance, in the example above, the report could show that a particular CSC in the STANDBY state will transition to the DISABLED state after receiving a ``start`` command and will also publish a series of events in the process, with a particular expected payload.
+Furthermore, if a component is supposed to publish certain events and fail to do so, the test reports will mark the condition as a failure.
+
+
+.. _robot-framework:
+
+Robot Framework
+---------------
+
+`Robot Framework`_ is broadly used for test automation.
+In addition to being able to execute tests, it can also be used to generate detailed reports with clear pass/fail criteria.
+
+In our testing suite we mostly rely on Robot Framework to verify the test results and generate test reports.
+The tests themselves are driven by a separate service that is in charge of queueing the required SAL Scripts in the ScriptQueue.
+
+An example report generated by Robot Framework is available `here <_static/tts_cycle0026_20220629/report.html>`__.
+
+.. _Robot Framework: https://robotframework.org
+
+.. _activity-sequence-diagrams:
+
+Activity Sequence Diagrams
+--------------------------
+
```

Comment:
You didn't really explain what the point of this section is.

Are these to generate artifacts? Why are they important for verification?

Reply:
Ok, I did some re-wording to clarify...